### PR TITLE
spires_parser: spires query fix

### PR DESCRIPTION
**Pull request by Panos512**
- Closes inspirehep/inspire-next#732

Signed-off-by: Panos Paparrigopoulos panos.paparrigopoulos@cern.ch

### CODE REVIEW

**Comment by jmartinm** [comment]
:+1: 

What is the effect of doing AND with an empty value?

```
AndOp(SpiresOp(Keyword('t'), Value('quark')), EmptyQuery('')))
```

Will it return no results? Or the ones coming from the `quark` value?

**Comment by Panos512** [comment]
It will return the results of the query before `and`, so yeah in this specific example it will return the results of `find t quark`.

The default inspirehep behavior was to return zero results (i suspect instead of an exception it just returns zero results).

**Comment by jmartinm** [comment]
I did not check verbose to see the query translation but does not seem to return 0 results?

https://inspirehep.net/search?ln=en&ln=en&p=find+t+quark+and&of=hb&action_search=Search&sf=earliestdate&so=d&rm=&rg=25&sc=0

**Comment by Panos512** [comment]
That's strange. I was checking with `find a Ellis` and `find a Ellis and` which returns 3.132 and 0 results respectively.

File: invenio_query_parser/contrib/spires/parser.py [hunk 1]
```diff
@@ -172,6 +172,7 @@ class SpiresAndQuery(UnaryRule):
             (omit(Whitespace), attr('op', SpiresSimpleQuery)),
             (omit(_), attr('op', SpiresParenthesizedQuery)),
             (omit(Whitespace), attr('op', SpiresValueQuery)),
+            (omit(re.compile(r".*", re.I)), attr('op', EmptyQueryRule)),
```
Comment (kaplun) [inline invenio_query_parser/contrib/spires/parser.py:175]: what does this do?

**Comment by kaplun** [comment]
The behavior on legacy was to consider "and" as part of the title. 

```
basic search units are: [['+', 'quark', 'title', 'w'], ['+', 'and', 'title', 'w']]
```

I don't think we should go too deep into this. The important thing is not to crash... ever! :dancer:

**Pull request merged**